src/realm/object-store/shared_realm.cpp

```diff
@@ -731,17 +744,18 @@ void Realm::run_writes()
         if (new_version > prev_version) {
             // A commit was done during callback
             --run_limit;
-            if (!run_limit)
-                m_async_commit_barrier_requested = true;
+            if (run_limit <= 0)
```

Comment:
Would it be better to reset the run_limit immediately?

Reply:
Not sure what you mean. We are breaking the loop and run_limit is not used anymore.

Reply:
my bad.